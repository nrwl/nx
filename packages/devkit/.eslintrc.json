{
  "extends": "../../.eslintrc.json",
  "ignorePatterns": ["!**/*", "node_modules"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {
        "@typescript-eslint/explicit-module-boundary-types": "error",
        "@typescript-eslint/no-restricted-imports": [
          "error",
          {
            "paths": [
              "@nx/workspace",
              "@angular-devkit/core",
              "@angular-devkit/architect",
              "@angular-devkit/schematics"
            ],
            "patterns": [
              {
                "group": [
                  "nx/bin/*",
                  "nx/src/*",
                  "nx/plugins/*",
                  "!nx/src/devkit-internals",
                  "!nx/src/devkit-exports"
                ],
                "allowTypeImports": true,
                "message": "Only import from nx/src/devkit-internals or nx/src/devkit-exports"
              },
              {
                "group": ["@nx/devkit/**/*"],
                "message": "Use a relative import"
              }
            ]
          }
        ]
      }
    },
    {
      "files": ["*.spec.ts"],
      "rules": {
        "@typescript-eslint/no-restricted-imports": "off"
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["./package.json"],
      "parser": "jsonc-eslint-parser",
      "rules": {
        "@nx/nx-plugin-checks": "error"
      }
    },
    {
      "files": ["./package.json"],
      "parser": "jsonc-eslint-parser",
      "rules": {
        "@nx/dependency-checks": [
          "error",
          {
            "buildTargets": ["build-base"],
            "ignoredDependencies": [
              "nx",
              "typescript",
              "prettier",
              "tslib",
              // Installed to workspace by plugins
              "rxjs",
              "@angular-devkit/core",
              "@angular-devkit/architect",
              "@angular-devkit/schematics",
              "webpack"
            ]
          }
        ]
      }
    }
  ]
}
