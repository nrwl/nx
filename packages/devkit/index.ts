/**
 * The Nx Devkit is the underlying technology used to customize Nx to support
 * different technologies and custom use-cases. It contains many utility
 * functions for reading and writing files, updating configuration,
 * working with Abstract Syntax Trees(ASTs), and more.
 *
 * As with most things in Nx, the core of Nx Devkit is very simple.
 * It only uses language primitives and immutable objects
 * (the tree being the only exception).
 *
 * @module @nx/devkit
 */

// eslint-disable-next-line @typescript-eslint/no-restricted-imports
export * from 'nx/src/devkit-exports';
export * from './public-api';
