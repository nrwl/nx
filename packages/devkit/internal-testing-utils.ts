/* eslint-disable @nx/enforce-module-boundaries */
export * from '../nx/src/internal-testing-utils/assert-valid-migrations';
export * from '../nx/src/internal-testing-utils/run-migration-against-this-workspace';
export * from '../nx/src/internal-testing-utils/with-environment';
export * from '../nx/src/internal-testing-utils/temp-fs';
/* eslint-enable @nx/enforce-module-boundaries */
export { setCwd } from './src/generators/artifact-name-and-directory-utils';
