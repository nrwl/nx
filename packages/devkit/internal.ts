export { signalToCode } from 'nx/src/devkit-internals';
