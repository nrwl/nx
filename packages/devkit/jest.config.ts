/* eslint-disable */
export default {
  transform: {
    '^.+\\.[tj]sx?$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  globals: {},
  displayName: 'cli',
  preset: '../../jest.preset.js',
  moduleNameMapper: {
    // Map Angular schematics to node_modules
    '^@schematics/angular/collection.json$':
      '<rootDir>/../../node_modules/@schematics/angular/collection.json',
    '^@angular-devkit/schematics/tools$':
      '<rootDir>/../../node_modules/@angular-devkit/schematics/tools/index.js',
  },
};
