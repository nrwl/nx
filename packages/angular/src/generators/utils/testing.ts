import type { Tree } from '@nx/devkit';
import { names, readProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Linter } from '@nx/eslint';
import { UnitTestRunner } from '../../utils/test-runners';
import { applicationGenerator } from '../application/application';
import type { Schema as ApplicationOptions } from '../application/schema';
import { componentGenerator } from '../component/component';
import { host } from '../host/host';
import type { Schema as HostOptions } from '../host/schema';
import { libraryGenerator } from '../library/library';
import type { Schema as LibraryOptions } from '../library/schema';
import { remote } from '../remote/remote';
import type { Schema as RemoteOptions } from '../remote/schema';

export async function generateTestApplication(
  tree: Tree,
  options: ApplicationOptions
): Promise<void> {
  tree.write('.gitignore', '');
  await applicationGenerator(tree, {
    ...options,
  });
}

export async function generateTestHostApplication(
  tree: Tree,
  options: HostOptions
): Promise<void> {
  tree.write('.gitignore', '');
  await host(tree, { ...options });
}

export async function generateTestRemoteApplication(
  tree: Tree,
  options: RemoteOptions
): Promise<void> {
  tree.write('.gitignore', '');
  await remote(tree, { ...options });
}

export async function generateTestLibrary(
  tree: Tree,
  options: LibraryOptions
): Promise<void> {
  tree.write('.gitignore', '');
  await libraryGenerator(tree, {
    ...options,
  });
}

export async function createStorybookTestWorkspaceForLib(
  libName: string
): Promise<Tree> {
  let tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
  tree.write('.gitignore', '');

  await libraryGenerator(tree, {
    directory: libName,
    buildable: false,
    linter: Linter.EsLint,
    publishable: false,
    simpleName: false,
    skipFormat: true,
    unitTestRunner: UnitTestRunner.Jest,
    standalone: false,
  });

  await componentGenerator(tree, {
    name: 'test-button',
    path: `${libName}/src/lib/test-button/test-button`,
    standalone: false,
    skipFormat: true,
  });

  tree.write(
    `${libName}/src/lib/test-button/test-button.component.ts`,
    `import { Component, Input } from '@angular/core';

export type ButtonStyle = 'default' | 'primary' | 'accent';

@Component({
  selector: 'proj-test-button',
  templateUrl: './test-button.component.html',
  styleUrls: ['./test-button.component.css']
})
export class TestButtonComponent {
  @Input('buttonType') type = 'button';
  @Input() style: ButtonStyle = 'default';
  @Input() age?: number;
  @Input() isOn = false;
}`
  );

  tree.write(
    `${libName}/src/lib/test-button/test-button.component.html`,
    `<button [attr.type]="type" [ngClass]="style"></button>`
  );

  const modulePath = `${libName}/src/lib/${libName}.module.ts`;
  tree.write(
    modulePath,
    `import * as ButtonExports from './test-button/test-button.component';
    ${tree.read(modulePath)}`
  );

  // create a module with component that gets exported in a barrel file
  generateModule(tree, {
    name: 'barrel',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'barrel-button',
    path: `${libName}/src/lib/barrel/barrel-button/barrel-button`,
    module: 'barrel',
    standalone: false,
    skipFormat: true,
  });

  tree.write(
    `${libName}/src/lib/barrel/barrel-button/index.ts`,
    `export * from './barrel-button.component';`
  );

  tree.write(
    `${libName}/src/lib/barrel/barrel.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BarrelButtonComponent } from './barrel-button';

@NgModule({
  imports: [CommonModule],
  declarations: [BarrelButtonComponent],
})
export class BarrelModule {}`
  );

  // create a module with components that get Angular exported and declared by variable
  generateModule(tree, {
    name: 'variable-declare',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'variable-declare-button',
    path: `${libName}/src/lib/variable-declare/variable-declare-button/variable-declare-button`,
    module: 'variable-declare',
    standalone: false,
    skipFormat: true,
  });

  await componentGenerator(tree, {
    name: 'variable-declare-view',
    path: `${libName}/src/lib/variable-declare/variable-declare-view/variable-declare-view`,
    module: 'variable-declare',
    standalone: false,
    skipFormat: true,
  });

  tree.write(
    `${libName}/src/lib/variable-declare/variable-declare.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VariableDeclareButtonComponent } from './variable-declare-button/variable-declare-button.component';
import { VariableDeclareViewComponent } from './variable-declare-view/variable-declare-view.component';

const COMPONENTS = [
  VariableDeclareButtonComponent,
  VariableDeclareViewComponent
]

@NgModule({
  imports: [CommonModule],
  declarations: COMPONENTS,
  exports: COMPONENTS
})
export class VariableDeclareModule {}`
  );

  // create a module with components that get Angular exported and declared by variable
  generateModule(tree, {
    name: 'variable-spread-declare',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'variable-spread-declare-button',
    path: `${libName}/src/lib/variable-spread-declare/variable-spread-declare-button/variable-spread-declare-button`,
    module: 'variable-spread-declare',
    standalone: false,
    skipFormat: true,
  });

  await componentGenerator(tree, {
    name: 'variable-spread-declare-view',
    path: `${libName}/src/lib/variable-spread-declare/variable-spread-declare-view/variable-spread-declare-view`,
    module: 'variable-spread-declare',
    standalone: false,
    skipFormat: true,
  });

  await componentGenerator(tree, {
    name: 'variable-spread-declare-anotherview',
    path: `${libName}/src/lib/variable-spread-declare/variable-spread-declare-anotherview/variable-spread-declare-anotherview`,
    module: 'variable-spread-declare',
    standalone: false,
    skipFormat: true,
  });

  tree.write(
    `${libName}/src/lib/variable-spread-declare/variable-spread-declare.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VariableSpreadDeclareButtonComponent } from './variable-spread-declare-button/variable-spread-declare-button.component';
import { VariableSpreadDeclareViewComponent } from './variable-spread-declare-view/variable-spread-declare-view.component';
import { VariableSpreadDeclareAnotherviewComponent } from './variable-spread-declare-anotherview/variable-spread-declare-anotherview.component';

const COMPONENTS = [
  VariableSpreadDeclareButtonComponent,
  VariableSpreadDeclareViewComponent
]

@NgModule({
  imports: [CommonModule],
  declarations: [...COMPONENTS, VariableSpreadDeclareAnotherviewComponent],
})
export class VariableSpreadDeclareModule {}`
  );

  // create a module where declared components are pulled from a static member of the module
  generateModule(tree, {
    name: 'static-member-declarations',
    project: libName,
  });

  await componentGenerator(tree, {
    name: 'cmp1',
    path: `${libName}/src/lib/static-member-declarations/cmp1/cmp1`,
    module: 'static-member-declarations',
    standalone: false,
    skipFormat: true,
  });

  await componentGenerator(tree, {
    name: 'cmp2',
    path: `${libName}/src/lib/static-member-declarations/cmp2/cmp2`,
    module: 'static-member-declarations',
    standalone: false,
    skipFormat: true,
  });

  tree.write(
    `${libName}/src/lib/static-member-declarations/static-member-declarations.module.ts`,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Cmp1Component } from './cmp1/cmp1.component';
import { Cmp2Component } from './cmp2/cmp2.component';

@NgModule({
  imports: [CommonModule],
  declarations: StaticMemberDeclarationsModule.COMPONENTS,
  exports: StaticMemberDeclarationsModule.COMPONENTS
})
export class StaticMemberDeclarationsModule {
  static readonly COMPONENTS = [Cmp1Component, Cmp2Component];
}`
  );

  // create another button in a nested subpath
  generateModule(tree, {
    name: 'nested',
    project: libName,
    path: `${libName}/src/lib`,
  });

  await componentGenerator(tree, {
    name: 'nested-button',
    module: 'nested',
    path: `${libName}/src/lib/nested/nested-button/nested-button`,
    standalone: false,
    skipFormat: true,
  });

  await componentGenerator(tree, {
    name: 'test-other',
    path: `${libName}/src/lib/test-other/test-other`,
    standalone: false,
    skipFormat: true,
  });

  return tree;
}

function generateModule(
  tree: Tree,
  options: { name: string; project: string; path?: string }
): void {
  const project = readProjectConfiguration(tree, options.project);

  if (options.path === undefined) {
    const sourceRoot = project.sourceRoot ?? `${project.root}/src`;
    const projectDirName =
      project.projectType === 'application' ? 'app' : 'lib';
    options.path = `${sourceRoot}/${projectDirName}`;
  }

  const moduleNames = names(options.name);
  const moduleFilePath = `${options.path}/${moduleNames.fileName}/${moduleNames.fileName}.module.ts`;

  tree.write(
    moduleFilePath,
    `import { NgModule } from '@angular/core';
  import { CommonModule } from '@angular/common';

  @NgModule({
    declarations: [],
    imports: [CommonModule],
  })
  export class ${moduleNames.className}Module {}`
  );
}
