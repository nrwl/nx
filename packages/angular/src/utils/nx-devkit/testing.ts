import type { Tree } from '@nx/devkit';
import { addProjectConfiguration, names, updateJson } from '@nx/devkit';

export interface AppConfig {
  appName: string; // name of app
  appModule: string; // app/app.module.ts in the above sourceDir
}

export interface LibConfig {
  name: string;
  module: string;
  barrel: string;
}

var appConfig: AppConfig; // configure built in createApp()
var libConfig: LibConfig;

export function getAppConfig(): AppConfig {
  return appConfig;
}

export function getLibConfig(): LibConfig {
  return libConfig;
}

export function createApp(
  tree: Tree,
  appName: string,
  routing: boolean = true
) {
  appName = names(appName).fileName;
  // save for getAppDir() lookup by external *.spec.ts tests
  appConfig = {
    appName,
    appModule: `${appName}/src/app/app.module.ts`,
  };

  tree.write(
    appConfig.appModule,
    `import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
${routing ? "import { RouterModule } from '@angular/router'" : ''};
import { App } from './app';
@NgModule({
  imports: [BrowserModule, ${routing ? 'RouterModule.forRoot([])' : ''}],
  declarations: [App],
  bootstrap: [App]
})
export class AppModule {}
`
  );
  tree.write(
    `${appName}/src/main.ts`,
    `import { platformBrowser } from '@angular/platform-browser';

import { AppModule } from './app/app.module';

platformBrowser()
  .bootstrapModule(AppModule)
  .catch(err => console.log(err));
  `
  );
  tree.write(
    `${appName}/tsconfig.app.json`,
    JSON.stringify({
      include: ['**/*.ts'],
    })
  );
  tree.write(
    `${appName}-e2e/tsconfig.e2e.json`,
    JSON.stringify({
      include: ['../**/*.ts'],
    })
  );
  addProjectConfiguration(tree, appName, {
    root: `${appName}`,
    sourceRoot: `${appName}/src`,
    targets: {
      build: {
        options: {
          main: `${appName}/src/main.ts`,
        },
      },
      serve: {
        options: {},
      },
    },
    tags: [],
  });
}

export function createLib(tree: Tree, libName: string) {
  const { name, className, fileName, propertyName } = names(libName);

  libConfig = {
    name,
    module: `${propertyName}/src/lib/${fileName}.module.ts`,
    barrel: `${propertyName}/src/index.ts`,
  };

  tree.write(
    libConfig.module,
    `import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
@NgModule({
  imports: [
    CommonModule
  ],
  providers: []
})
export class ${className}Module { }
  `
  );
  tree.write(
    libConfig.barrel,
    `export * from './lib/${fileName}.module';
`
  );
}
