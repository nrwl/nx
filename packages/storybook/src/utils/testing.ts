import { Tree } from '@nx/devkit';

import { libraryGenerator } from '@nx/js';
import { Linter } from '@nx/eslint';

export async function createTestUILibNoNgDevkit(
  appTree: Tree,
  libName: string
): Promise<Tree> {
  await libraryGenerator(appTree, {
    linter: Linter.EsLint,
    skipFormat: true,
    skipTsConfig: false,
    unitTestRunner: 'none',
    directory: libName,
  });

  return appTree;
}

export function deleteNewConfigurationAndCreateNew(
  appTree: Tree,
  projectStorybookRoot: string
): Tree {
  // Remove new Storybook configuration
  appTree.delete(`.storybook/main.js`);
  appTree.delete(`.storybook/tsconfig.json`);
  appTree.delete(`${projectStorybookRoot}/main.js`);
  appTree.delete(`${projectStorybookRoot}/preview.js`);
  appTree.delete(`${projectStorybookRoot}/tsconfig.json`);

  // Create old Storybook configuration
  appTree.write(`.storybook/addons.js`, 'console.log("hello")');
  appTree.write(`.storybook/webpack.config.js`, 'console.log("hello")');
  appTree.write(`.storybook/tsconfig.json`, '{"test": "hello"}');
  appTree.write(`${projectStorybookRoot}/config.js`, 'console.log("hello")');
  appTree.write(`${projectStorybookRoot}/addons.js`, 'console.log("hello")');
  appTree.write(
    `${projectStorybookRoot}/webpack.config.js`,
    'console.log("hello")'
  );
  appTree.write(`${projectStorybookRoot}/tsconfig.json`, '{"test": "hello"}');

  return appTree;
}
