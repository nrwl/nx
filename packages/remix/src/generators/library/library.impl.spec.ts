import 'nx/src/internal-testing-utils/mock-project-graph';

import {
  readJson,
  readProjectConfiguration,
  Tree,
  updateJson,
  writeJson,
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import applicationGenerator from '../application/application.impl';
import libraryGenerator from './library.impl';

describe('Remix Library Generator', () => {
  it('should generate a library correctly', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

    // ACT
    await libraryGenerator(tree, {
      directory: 'test',
      style: 'css',
      addPlugin: true,
    });

    // ASSERT
    const tsconfig = readJson(tree, 'tsconfig.base.json');
    expect(tree.exists(`test/src/server.ts`));
    expect(tree.children(`test/src/lib`).sort()).toMatchSnapshot();
    expect(tsconfig.compilerOptions.paths).toMatchSnapshot();
  }, 25_000);

  describe('Standalone Project Repo', () => {
    it('should update the tsconfig paths correctly', async () => {
      // ARRANGE
      const tree = createTreeWithEmptyWorkspace();
      await applicationGenerator(tree, {
        name: 'demo',
        directory: '.',
        rootProject: true,
        addPlugin: true,
      });
      const originalBaseTsConfig = readJson(tree, 'tsconfig.json');

      // ACT
      await libraryGenerator(tree, {
        directory: 'test',
        style: 'css',
        addPlugin: true,
      });

      // ASSERT
      const updatedBaseTsConfig = readJson(tree, 'tsconfig.base.json');
      expect(Object.keys(originalBaseTsConfig.compilerOptions.paths)).toContain(
        '~/*'
      );
      expect(Object.keys(updatedBaseTsConfig.compilerOptions.paths)).toContain(
        '~/*'
      );
    });
  });

  describe('--unitTestRunner', () => {
    it('should not create config files when unitTestRunner=none', async () => {
      // ARRANGE
      const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

      // ACT
      await libraryGenerator(tree, {
        directory: 'test',
        style: 'css',
        unitTestRunner: 'none',
        addPlugin: true,
      });

      // ASSERT
      expect(tree.exists(`test/jest.config.ts`)).toBeFalsy();
      expect(tree.exists(`test/vite.config.ts`)).toBeFalsy();
    });

    it('should create the correct config files for testing with jest', async () => {
      // ARRANGE
      const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

      // ACT
      await libraryGenerator(tree, {
        directory: 'test',
        style: 'css',
        unitTestRunner: 'jest',
        addPlugin: true,
      });

      // ASSERT
      expect(tree.read(`test/jest.config.ts`, 'utf-8')).toMatchSnapshot();
      expect(tree.read(`test/src/test-setup.ts`, 'utf-8')).toMatchSnapshot();
    });

    it('should create the correct config files for testing with vitest', async () => {
      // ARRANGE
      const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

      // ACT
      await libraryGenerator(tree, {
        directory: 'test',
        style: 'css',
        unitTestRunner: 'vitest',
        addPlugin: true,
      });

      // ASSERT
      expect(tree.read(`test/vite.config.ts`, 'utf-8')).toMatchSnapshot();

      expect(tree.read(`test/src/test-setup.ts`, 'utf-8')).toMatchSnapshot();
    }, 25_000);
  });

  // TODO(Colum): Unskip this when buildable is investigated correctly
  xit('should generate the config files correctly when the library is buildable', async () => {
    // ARRANGE
    const tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });

    // ACT
    await libraryGenerator(tree, {
      directory: 'test',
      style: 'css',
      buildable: true,
      addPlugin: true,
    });

    // ASSERT
    const project = readProjectConfiguration(tree, 'test');
    const pkgJson = readJson(tree, `test/package.json`);
    expect(project.targets.build.options.format).toEqual(['cjs']);
    expect(project.targets.build.options.outputPath).toEqual(`test/dist`);
    expect(pkgJson.main).toEqual('./dist/index.cjs.js');
    expect(pkgJson.typings).toEqual('./dist/index.d.ts');
  });
  describe('TS solution setup', () => {
    let tree: Tree;

    beforeEach(() => {
      tree = createTreeWithEmptyWorkspace();
      updateJson(tree, 'package.json', (json) => {
        json.workspaces = ['packages/*', 'apps/*'];
        return json;
      });
      writeJson(tree, 'tsconfig.base.json', {
        compilerOptions: {
          composite: true,
          declaration: true,
        },
      });
      writeJson(tree, 'tsconfig.json', {
        extends: './tsconfig.base.json',
        files: [],
        references: [],
      });
    });

    it('should generate valid package.json', async () => {
      await libraryGenerator(tree, {
        directory: 'packages/foo',
        style: 'css',
        addPlugin: true,
      });

      expect(readJson(tree, 'packages/foo/package.json'))
        .toMatchInlineSnapshot(`
        {
          "main": "./src/index.ts",
          "name": "@proj/foo",
          "nx": {
            "name": "foo",
            "projectType": "library",
            "sourceRoot": "packages/foo/src",
          },
          "types": "./src/index.ts",
          "version": "0.0.1",
        }
      `);
    });
  });
});
