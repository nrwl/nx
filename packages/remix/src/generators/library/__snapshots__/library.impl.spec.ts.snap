// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remix Library Generator --unitTestRunner should create the correct config files for testing with jest 1`] = `
"export default {
  setupFilesAfterEnv: ['./src/test-setup.ts'],
  displayName: 'test',
  preset: '../jest.preset.js',
  transform: {
    '^(?!.*\\\\.(js|jsx|ts|tsx|css|json)$)': '@nx/react/plugins/jest',
    '^.+\\\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../coverage/test',
};
"
`;

exports[`Remix Library Generator --unitTestRunner should create the correct config files for testing with jest 2`] = `
"import { installGlobals } from '@remix-run/node';
import '@testing-library/jest-dom/matchers';
installGlobals();
"
`;

exports[`Remix Library Generator --unitTestRunner should create the correct config files for testing with vitest 1`] = `
"import { defineConfig } from 'vite';
import react from '@vitejs/plugin-react';
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import { nxCopyAssetsPlugin } from '@nx/vite/plugins/nx-copy-assets.plugin';

export default defineConfig(() => ({
  root: __dirname,
  cacheDir: '../node_modules/.vite/test',
  plugins: [react(), nxViteTsPaths(), nxCopyAssetsPlugin(['*.md'])],
  // Uncomment this if you are using workers.
  // worker: {
  //  plugins: [ nxViteTsPaths() ],
  // },
  test: {
    setupFiles: ['./src/test-setup.ts'],
    watch: false,
    globals: true,
    environment: 'jsdom',
    include: ['src/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
    reporters: ['default'],
    coverage: {
      reportsDirectory: '../coverage/test',
      provider: 'v8' as const,
    },
  },
}));
"
`;

exports[`Remix Library Generator --unitTestRunner should create the correct config files for testing with vitest 2`] = `
"import { installGlobals } from '@remix-run/node';
import '@testing-library/jest-dom/matchers';
installGlobals();
"
`;

exports[`Remix Library Generator should generate a library correctly 1`] = `
[
  "test.module.css",
  "test.spec.tsx",
  "test.tsx",
]
`;

exports[`Remix Library Generator should generate a library correctly 2`] = `
{
  "@proj/test": [
    "test/src/index.ts",
  ],
  "@proj/test/server": [
    "test/src/server.ts",
  ],
}
`;
